internal/fetcher/fetcher_test.go

```diff
@@ -142,12 +150,16 @@ func TestFetchSameKeysSlow(t *testing.T) {
 			stateKeys.Add(keyBase+strconv.Itoa(k), state.Read)
 		}
 		txID := ids.GenerateTestID()
+		keys := make([]string, 0, len(stateKeys))
```

Comment:
converting the state keys map into a keys slice seems to be a common usage pattern; do you think it would make sense to create a function that would encapsulate that ?

Reply:
For the tests, we can switch to generating database keys instead of generating state keys + immediately stripping their permissions. But agree that we should add a function that converts state keys into a keys slice.

Reply:
Done